### ISSUE TITLE
Encountering ArrayIndexOutOfBoundsException on IonQuant step

### ISSUE THREAD

**Issue by turtlypo** [Nesvilab/FragPipe#501]
Hello FragPipe team!

I'm running FragPipe v16.0 with msFragger v3.3, and Philosopher v4.0.0.  I've been trying to search timsTOF .d files against a .fasta.fas database (I used the GUI to add decoys and contaminants).  I also turned on peptideProphet and proteinProphet, and enabled MS1 quant via IonQuant.  

On the console, I've been encountering an error during the IonQuant stage, right as the system starts to update Philosopher tables.  

```
2021-10-20 13:32:17 [INFO] - Quantifying...
2021-10-20 13:32:17 [INFO] - Updating Philosopher's tables...
java.lang.ArrayIndexOutOfBoundsException: -1
	at com.univocity.parsers.common.DefaultContext.selectedHeaders(DefaultContext.java:62)
	at com.univocity.parsers.common.record.RecordMetaDataImpl.selectedHeaders(RecordMetaDataImpl.java:213)
	at com.univocity.parsers.common.record.RecordMetaDataImpl.getMetaData(RecordMetaDataImpl.java:50)
	at com.univocity.parsers.common.record.RecordMetaDataImpl.metadataOf(RecordMetaDataImpl.java:114)
	at com.univocity.parsers.common.record.RecordMetaDataImpl.getObjectValue(RecordMetaDataImpl.java:374)
	at com.univocity.parsers.common.record.RecordImpl.getString(RecordImpl.java:113)
	at o.<init>(Unknown Source)
	at ionquant.IonQuant.main(Unknown Source)
2021-10-20 13:32:18 [ERROR] - -1
```

[Here's a compressed folder of logs from my last three erroneous runs](https://github.com/Nesvilab/FragPipe/files/7383790/Logs.zip)

1. Log _____18-50-35.txt was the first time I encountered this error.
2. For log _____12-40-01.txt, I unchecked msFragger and Validation tasks and only re-ran IonQuant.
3. In the most recent log (13-32-25.txt), I kept msFragger unchecked but also re-ran peptide and proteinProphets to see if that helped.  

To my knowledge Philosopher appears to be up-to-date.  Is there something regarding my configuration or Validation step that's causing this?

I'm not sure if this is a connected issue, but the previous day I tried to run the same workflow with RAM set to 32 and parallelism set to 48, but received a Java OutOfMemory error.  Setting RAM to 0 resolved this.

**Comment by fcyu** [comment]
I think it should be due to the known issue (https://github.com/Nesvilab/FragPipe/issues/419). Please remove `E:\Active_Data\Daniel\14_LO6\14Oct2021_DDH_LO6E_Slot2-27_3_2557.d`, and try again.

Best,

Fengchao

**Comment by turtlypo** [comment]
Ahh yes, I completely forget to update the input files.  
[log_2021-10-20_17-03-53.txt.zip](https://github.com/Nesvilab/FragPipe/files/7384583/log_2021-10-20_17-03-53.txt.zip)
A similar thing occurred again.  I can try to remove another input file, but I'm not sure if it's just the last file it was quantifying to remove.  

```
2021-10-20 17:01:36 [INFO] - Building index...
2021-10-20 17:03:31 [INFO] - Writing index to E:\Active_Data\Daniel\14_LO6\14Oct2021_DDH_LO6A_Slot2-23_3_2545.quantindex
2021-10-20 17:03:36 [INFO] - Loading E:\Active_Data\Daniel\14_LO6\FraggerReRun\Control_3\14Oct2021_DDH_LO6A_Slot2-23_3_2545.pepXML...
2021-10-20 17:03:42 [INFO] - Use each MS2 scan's calculated MZ in peak tracing.
2021-10-20 17:03:42 [INFO] - Quantifying...
2021-10-20 17:03:42 [INFO] - Updating Philosopher's tables...
java.lang.ArrayIndexOutOfBoundsException: -1
	at com.univocity.parsers.common.DefaultContext.selectedHeaders(DefaultContext.java:62)
	at com.univocity.parsers.common.record.RecordMetaDataImpl.selectedHeaders(RecordMetaDataImpl.java:213)
	at com.univocity.parsers.common.record.RecordMetaDataImpl.getMetaData(RecordMetaDataImpl.java:50)
	at com.univocity.parsers.common.record.RecordMetaDataImpl.metadataOf(RecordMetaDataImpl.java:114)
	at com.univocity.parsers.common.record.RecordMetaDataImpl.getObjectValue(RecordMetaDataImpl.java:374)
	at com.univocity.parsers.common.record.RecordImpl.getString(RecordImpl.java:113)
	at o.<init>(Unknown Source)
	at ionquant.IonQuant.main(Unknown Source)
2021-10-20 17:03:46 [ERROR] - -1
Process 'IonQuant' finished, exit code: 1
Process returned non-zero exit code, stopping

~~~~~~~~~~~~~~~~~~~~
Cancelling 2 remaining tasks

```

In this case, should I try removing `E:\Active_Data\Daniel\14_LO6\FraggerReRun\Control_3\14Oct2021_DDH_LO6A_Slot2-23_3_2545.d`?

**Comment by fcyu** [comment]
You need to start the job from scratch since you removed one run but the result files still have it.

**Comment by fcyu** [comment]
It is still due to the same reason. There are no PSM from your 156x_1.
```
PhilosopherFilter [Work dir: E:\Active_Data\Daniel\NewLO6Run\FraggerOutput2\156x_1]
C:\AllMSFraggerFiles\fragpipe\tools\philosopher\philosopher.exe filter --sequential --razor --mapmods --prot 0.01 --tag rev_ --pepxml E:\Active_Data\Daniel\NewLO6Run\FraggerOutput2\156x_1 --protxml E:\Active_Data\Daniel\NewLO6Run\FraggerOutput2\combined.prot.xml --razorbin E:\Active_Data\Daniel\NewLO6Run\FraggerOutput2\13_25x_1\.meta\razor.bin
INFO[14:37:50] Executing Filter  v4.0.0                     
INFO[14:37:50] Processing peptide identification files      
WARN[14:37:50] No PSM was found in data set. interact.pep.xml 
INFO[14:37:50] 1+ Charge profile                             decoy=0 target=0
INFO[14:37:50] 2+ Charge profile                             decoy=0 target=0
INFO[14:37:50] 3+ Charge profile                             decoy=0 target=0
INFO[14:37:50] 4+ Charge profile                             decoy=0 target=0
INFO[14:37:50] 5+ Charge profile                             decoy=0 target=0
INFO[14:37:50] 6+ Charge profile                             decoy=0 target=0
INFO[14:37:50] Database search results                       ions=0 peptides=0 psms=0
INFO[14:37:50] Converged to 0.00 % FDR with 0 PSMs           decoy=0 threshold=10 total=0
INFO[14:37:50] Converged to 0.00 % FDR with 0 Peptides       decoy=0 threshold=10 total=0
INFO[14:37:50] Converged to 0.00 % FDR with 0 Ions           decoy=0 threshold=10 total=0
INFO[14:37:51] Protein inference results                     decoy=3006 target=3176
INFO[14:37:51] Converged to 2.44 % FDR with 41 Proteins      decoy=1 threshold=0.9846 total=42
INFO[14:37:51] Applying sequential FDR estimation            ions=0 peptides=0 psms=0
INFO[14:37:51] Converged to 0.00 % FDR with 0 PSMs           decoy=0 threshold=10 total=0
INFO[14:37:51] Converged to 0.00 % FDR with 0 Peptides       decoy=0 threshold=10 total=0
INFO[14:37:51] Converged to 0.00 % FDR with 0 Ions           decoy=0 threshold=10 total=0
INFO[14:37:51] Post processing identifications              
INFO[14:37:52] Mapping modifications                        
INFO[14:37:52] Assigning protein identifications to layers  
INFO[14:37:52] Processing protein inference                 
INFO[14:37:53] Synchronizing PSMs and proteins              
INFO[14:37:53] Saving                                       
INFO[14:37:53] Done      
```

It looks like removing `14Oct2021_DDH_LO6E_Slot2-27_3_2557.d` causes some differences in the FDR threshold, which makes `14Oct2021_DDH_LO6E_Slot2-27_1_2555.d` have 0 PSMs (it got 5 PSMs before). 

You can actually remove all `156x_1`, `156x_2`, and `156x_3` since there are almost no PSM. I am not sure why there are no PSM in those three experiment, though.

Best,

Fengchao

**Comment by fcyu** [comment]
Fixed in FragPipe 17.0.

**Comment by turtlypo** [comment]
Sorry I forgot to update what happened! After removing all of the `156x` technical replicates, the run was successful.  Glad to hear this is fixed in v17.0, thank you for all the help!
